android/festago/app/src/main/java/com/festago/festago/data/repository/AuthDefaultRepository.kt

```diff
@@ -28,4 +31,29 @@ class AuthDefaultRepository(
             return Result.failure(e)
         }
     }
+
+    override suspend fun signOut(): Result<Unit> {
+        return try {
+            authDataSource.token = null
+            UserApiClient.instance.logout {}
```

Comment:
```suggestion
            UserApiClient.instance.logout {}
            authDataSource.token = null
```
로직 순서를 위로 바꾸는건 어떤가요??
`UserApiClient.instance.logout{}`이 실패하고 token이 null로 바뀌는 경우가 있을 것같아요!

Reply:
`로그아웃은 요청 성공 여부와 관계없이 토큰을 삭제 처리한다는 점에 유의합니다.` 라고 [문서에](https://developers.kakao.com/docs/latest/ko/kakaologin/android#logout) 나와있어서 실패하더라도 토큰을 null 처리하는 게 맞다고 생각했어요! 어떻게 생각하시나요?

Reply:
`logout` 같은 경우는 모든 네트워크 에러 포함해서 토큰을 삭제하고 있네요! 말씀해주신대로 null처리 우선인게 맞는거 같습니다!
<img width="650" alt="스크린샷 2023-08-10 오후 1 29 54" src="https://github.com/woowacourse-teams/2023-festa-go/assets/37167652/8f481379-b002-46fa-9f79-45c7375280bd">

`unlink` 같은 경우는 아래와 같이 error가 null일 경우에 토큰을 삭제 하고 있습니다! unlink는 순서를 변경해주는건 어떠신가요??
<img width="645" alt="스크린샷 2023-08-10 오후 1 28 47" src="https://github.com/woowacourse-teams/2023-festa-go/assets/37167652/63f978e6-9c38-4766-9c08-3969e924f786">

Reply:
지금보니 callback에 넣어줘도 괜찮을 것같네요!
```suggestion
            UserApiClient.instance.logout {
                    authDataSource.token = null
            }
```

Reply:
콜백 반영했습니다. 다만 토큰 null 처리 순서를 바꾸는거나 콜백으로 처리하는 게 의미가 있나요? 사실 signOut()은 try-catch도 의미없는 것 같기도..

Reply:
말씀해주신대로 signOut 같은 경우는 kakao api가 throw를 하지 않기 때문에 try-catch가 하는 역할이 없어서 지워도 문제 없을 것같아요! 콜백에서 에러를 처리하면 될 것같아요!

kakao api 모든 곳에서 try-catch를 지워도 될 것같습니다!

하지만 unlink 같은 경우는 저희 서버와 카카오 서버랑 토큰을 동시에 관리하는게 좋겠다고 생각해서 말씀드렸습니다!
카카오 서버 token 이랑 저희 서버 token이랑 관련이 없다고 보시나요??
```
UserApiClient.instance.unlink {
    if (error == null) {
        authDataSource.token = null
    }
}
```

Reply:
unlink의 경우 error가 없을 때 token 제거해야한다는 점 동의합니다! 반영했어요.